### PR TITLE
Alpha version

### PR DESCRIPTION
**Pull request by muriloviana**
Initial implementation for mongodex.

## How to test

Checkout this branch, create and activate a virtualenv, and run `python -m pip install -e .` to install mongodex in editable mode.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,42 @@
...
+```python
+mongodex.migrate("<DATABASE_URI>", collections)
+```
+
+## Markdown docs generator
+
+You can also create a markdown file with all your indexes using the same collection index dictionary mentioned in the previous topic. For example:
+
+```python
+mongodex.md_generator(collenctions)
```
Comment (mawkee) [inline README.md]:
```suggestion
mongodex.md_generator(collections)
```

File: mongodex/helpers.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            "keys": self.fields.items(),
+            "unique": self.unique,
+            "sparse": self.sparse,
+        }
+        if self.partial:
+            obj.update({"partialFilterExpression": self.partial})
+        if self.expire is not None:
+            obj.update({"expireAfterSeconds": self.expire})
+        # If we have at least one of 2dsphere, then it's a geospatial index
+        if GEOSPHERE in self.fields.values():
```
Comment (bordax) [inline mongodex/helpers.py:53]: Can a index be geospatial and textual at the same time?

File: mongodex/helpers.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            obj.update({"2dsphereIndexVersion": self.geosphere_index_version})
+        # If we have at least one field of type text, then it's a text index
+        if TEXT in self.fields.values():
+            obj.update(
+                {
+                    "language_override": self.language_override,
+                    "textIndexVersion": self.text_index_version,
+                    "weights": self.weights,
+                }
+            )
```
Comment (bordax) [inline mongodex/helpers.py:63]: Can a index be geospatial and textual at the same time?

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+authors = [
+    { name="Danilo Martins", email="mawkee@gmail.com" },
+    { name="Evandro Arruda", email="ecarruda@gmail.com" },
+    { name="Murilo Viana", email="murilo.vianamo@gmail.com" },
+]
+license = { file="LICENSE" }
+classifiers = [
+    "Programming Language :: Python :: 3",
+    "License :: OSI Approved :: MIT License",
+    "Operating System :: OS Independent",
```
Comment (bordax) [inline pyproject.toml:19]:
```suggestion
    "Operating System :: OS Independent",
    "Topic :: Database",
```

File: mongodex/helpers.py (hunk 3 above)
Comment (muriloviana) [reply mongodex/helpers.py:63]: Good question! Perhaps a compound index?

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+]
+requires-python = ">=3.7"
+dependencies = [
+    "loguru",
+    "pymongo",
+]
+
+[project.urls]
+"Homepage" = "https://github.com/pinio/mongodex"
+"Bug Tracker" = "https://github.com/pinio/mongodex/issues"
```
Comment (ecarruda) [inline pyproject.toml]: new line on end of file?!!

**Review by bordax — APPROVED** [review]

**Review by mawkee — APPROVED** [review]

File: mongodex/helpers.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+                name=other["name"],
+                comment=other.get("comment", ""),
+                unique=other.get("unique", False),
+                sparse=other.get("sparse", False),
+                partial=other.get("partialFilterExpression", None),
+                expire=other.get("expireAfterSeconds", None),
+                geosphere_index_version=other.get("2dsphereIndexVersion", 2),
+                language_override=other.get("language_override", "language"),
+                text_index_version=other.get("textIndexVersion", 3),
+                weights=other.get("weights", None),
```
Comment (ecarruda) [inline mongodex/helpers.py]: Are you setting None just to be explicit?

**Review by ecarruda — APPROVED** [review]
Congrats @muriloviana !

**Pull request merged**